packages/schema/src/JSONSchema.ts

```diff
@@ -526,3 +651,279 @@ const go = (ast: AST.AST, $defs: Record<string, JsonSchema7>): JsonSchema7 => {
+    // Handle enums
+    // ---------------------------------------------
+    Match.when(
+      isJsonSchema7Enum,
+      ({ enum: enum_ }) => Schema.union(...ReadonlyArray.map(enum_, (a) => Schema.literal(a)))
+    ),
+    Match.when(
+      isJsonSchema7Enums,
+      ({ oneOf }) => Schema.enums(Object.assign({}, ...oneOf.map(({ const: const_, title }) => ({ [title]: const_ }))))
+    ),
+    // ---------------------------------------------
+    // Hndle numberic types
+    // ---------------------------------------------
+    Match.whenOr(
+      isJsonSchema7Number,
+      isJsonSchema7Integer,
+      ({ exclusiveMaximum: exclusiveMax, exclusiveMinimum: exclusiveMin, maximum: max, minimum: min }) => {
+        const number = isJsonSchema7Integer(input) ? Schema.int()(Schema.number) : Schema.number
+        const lessThanOrEqualTo = Predicate.isNullable(min) ? Function.identity : Schema.greaterThanOrEqualTo(min)
+        const greaterThanOrEqualTo = Predicate.isNullable(max) ? Function.identity : Schema.lessThanOrEqualTo(max)
+        const lessThan = Predicate.isNullable(exclusiveMin) ? Function.identity : Schema.greaterThan(exclusiveMin)
+        const greaterThan = Predicate.isNullable(exclusiveMax) ? Function.identity : Schema.lessThan(exclusiveMax)
+        return Function.flow(lessThanOrEqualTo, greaterThanOrEqualTo, lessThan, greaterThan)(number)
+      }
+    ),
+    // ---------------------------------------------
+    // Handle string types
+    // ---------------------------------------------
+    Match.when(isJsonSchema7String, ({ description, maxLength, minLength, pattern }) => {
+      const withMaxLength = Predicate.isNullable(maxLength) ? Function.identity : Schema.maxLength(maxLength)
+      const withMinLength = Predicate.isNullable(minLength) ? Function.identity : Schema.minLength(minLength)
+      const withPattern = Predicate.isNullable(pattern) ? Function.identity : Schema.pattern(new RegExp(pattern))
+      const withDescription = Predicate.isNullable(description) ? Function.identity : Schema.description(description)
+      return Function.flow(withMaxLength, withMinLength, withPattern, withDescription)(Schema.string)
+    }),
+    // ---------------------------------------------
+    // Handle objects (will do recursive calls for the properties)
+    // ---------------------------------------------
+    Match.when(isJsonSchema7Object, ({ additionalProperties, patternProperties, properties, required }) => {
+      const fields = Object.entries(properties).map(([name, property]) => {
+        const isRequired = required.includes(name)
+        const decodedProperty = decodeWithReferences(property, references)
+        const withOptional = isRequired
+          ? Schema.required(decodedProperty)
+          : Schema.optional(decodedProperty, { exact: true }) // FIXME: How do I know if this is "exact" / will it always be "exact" here?
+        return [name, withOptional] as const
+      })
```

Comment:
@gcanti When decoding an object, I check the `required` array to see if it is required/optional, but how do I know the property is exact or not? Will it always be exact when decoding?